test/unit/test_reader.cpp

```diff
@@ -0,0 +1,28 @@
+#include "AtomicDataReader.hpp"
+#include <filesystem>
+#include <gtest/gtest.h>
+#include <string>
+#include <vector>
+
+TEST(ReadersTest, CSV) {
+
+  std::filesystem::path source_file = __FILE__;
+  std::filesystem::path source_dir = source_file.parent_path();
+  std::filesystem::path test_resources_dir = source_dir / "../test_resources";
+  std::filesystem::path csv_test_file =
+      test_resources_dir / "charge_exchange_h0_h1.csv";
+
+  std::vector<std::vector<double>> all_data = CSVReader(std::string(csv_test_file)).GetData();
+  std::vector<double> rates = CSVReader(std::string(csv_test_file)).GetRates();
+  std::vector<double> temps = CSVReader(std::string(csv_test_file)).GetTemps();
```

Comment:
This is creating three objects - better to just create one and call GetData(), GetRates() and GetTemps() on it

Reply:
Discussion moved to a separate conversation